services/m3dbnode/main/main.go

```diff
@@ -83,11 +84,20 @@ func main() {
 			dbOpts,
 			doneCh,
 		); err != nil {
-			log.Fatalf("serve error: %v", err)
+			log.Fatalf("server fatal error: %v", err)
 		}
+
+		// Server is closed
+		<-closedCh
 	}()
 
-	log.Fatalf("interrupt: %v", interrupt())
+	// Handle interrupt
+	log.Infof("interrupt: %v", interrupt())
+
+	// Attempt graceful server close
+	doneCh <- struct{}{}
+	<-closedCh
```

Comment:
Actually here I'm reading the channel to ensure the server has received the `doneCh` signal and has finished closing.